File: webApp/src/app/common/route.ts
import { Bus } from './bus';
import { Station } from './station';

/**
 * 车次实体
 */
export class Route {
  /** id */
  id: number;

  /** 车辆编号 */
  bus: Bus;

  /** 发车时间 */
  startTime: string;

  /** 到达时间 */
  endTime: string;

  /** 起始车站编号 */
  startStation: Station;

  /** 终点车站编号 */
  endStation: Station;

  /** 乘客数量 */
  buyNumber: number;

  /** 价格 */
  price: string;

  /** 折扣 */
  discount: number;

  constructor(data?: { id?: number, bus?: Bus, startTime?: string, endTime?: string, startStation?: Station,
    endStation?: Station, buyNumber?: number, price?: string, discount?: number }) {
    if (data) {
      this.id = data.id;
      this.bus = data.bus;
      this.startTime = data.startTime;
      this.endTime = data.endTime;
      this.startStation = data.startStation;
      this.endStation = data.endStation;
      this.buyNumber = data.buyNumber;
      this.price = data.price;
      this.discount = data.discount;
    }
  }
}

File: bus-ticket/src/main/java/xiaoqiangZzz/busticket/Entity/Route.java
package xiaoqiangZzz.busticket.Entity;


import com.mengyunzhi.core.entity.YunzhiEntity;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.Date;

@Entity
public class Route implements YunzhiEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Bus bus;

    private Date startTime;

    private Date endTime;

    @ManyToOne
    @JoinColumn(nullable = false)
    private City startCity;

    @ManyToOne
    @JoinColumn(nullable = false)
    private City endCity;


    @ManyToOne
    @JoinColumn(nullable = false)
    private Station startStation;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Station endStation;

    private int buyNumber;

    private float price;

    private float discount;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Bus getBus() {
        return bus;
    }

    public void setBus(Bus bus) {
        this.bus = bus;
    }

    public Station getStartStation() {
        return startStation;
    }

    public void setStartStation(Station startStation) {
        this.startStation = startStation;
    }

    public Station getEndStation() {
        return endStation;
    }

    public void setEndStation(Station endStation) {
        this.endStation = endStation;
    }

    public int getBuyNumber() {
        return buyNumber;
    }

    public void setBuyNumber(int buyNumber) {
        this.buyNumber = buyNumber;
    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public float getDiscount() {
        return discount;
    }

    public void setDiscount(float discount) {
        this.discount = discount;
    }

    public Date getStartTime() {
        return startTime;
    }

    public void setStartTime(Date startTime) {
        this.startTime = startTime;
    }

    public Date getEndTime() {
        return endTime;
    }

    public void setEndTime(Date endTime) {
        this.endTime = endTime;
    }

    public City getStartCity() {
        return startCity;
    }

    public void setStartCity(City startCity) {
        this.startCity = startCity;
    }

    public City getEndCity() {
        return endCity;
    }

    public void setEndCity(City endCity) {
        this.endCity = endCity;
    }
}
